packages/vite/src/node/plugins/css.ts

```diff
@@ -558,8 +558,7 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
         } else if (!config.build.ssr) {
           // legacy build and inline css
 
-          // __VITE_ASSET__ and __VITE_PUBLIC_ASSET__ urls are processed by
-          // the vite:asset plugin, don't call resolveAssetUrlsInCss here
+          chunkCSS = resolveAssetUrlsInCss(chunkCSS, chunk.name)
```

Comment:
Since we inject css by `document.head.appendChild`, I think we need to make the urls relative from `location.href`. (or an absolute url?)

Reply:
Good catch! I chose the plan of relative path. It would to keep the same with no legacy mode.

BTW, I found anther bug 👀  the main css chunk(it load by html) will load by js again in legacy mode. maybe it should be fix in other PR.